### ISSUE TITLE
Options broken in poudriere-devel 

### ISSUE THREAD

**Issue by mattxtaz** [freebsd/poudriere#1090]
## Prerequisites
- [X] Have you checked for an existing issue describing your problem?
- [X] Are you running the latest version?
- [X] Is your ports tree recent?
- [X] Is your FreeBSD Host on a [supported release](https://www.freebsd.org/security/#sup)?

## Describe the bug
When I run poudriere options it displays an error. This worked in the previous version and the issue started happening since the last port upgrade.

## How to reproduce
```
% poudriere options -n -f /usr/local/etc/poudriere.d/optlist
[00:00:00] Working on options directory: /usr/local/etc/poudriere.d/options
[00:00:00] Using ports from: /usr/ports
[00:00:00] Appending to make.conf: /usr/local/etc/poudriere.d/make.conf
[00:00:00] Error: _jget: 3 arguments expected: var_return ptname property
```

## Expected behavior
poudriere should loop through all of the ports in the options file to see if any of the options need to present a dialog to update them in the case of any options being added or removed.

## Screenshots
<If applicable add screenshots to help explain your problem.>

## Environment
 - Host OS [e.g. 12.2 amd64]: 13.2 amd64
 - Jail OS [e.g. 12.0 powerpc]: 13.2 amd64
 - Browser: [e.g. chrome, safari]: Firefox
 - Poudriere Version [e.g. 3.3.1 or git hash or port version]: poudriere-devel-3.4.99.20231113
 - Ports branch and revision [e.g. 2020Q3 r550754]: latest (259ddfbbd104e024361591fc5bec5a252c6eb5cb)

## Additional context
<Add any other context about the problem here.>

**Comment by mattxtaz** [comment]
Just adding another comment in case it's important. /usr/local/etc/poudriere.d/options is a symlink to /var/db/ports so that I can use make config in the /usr/ports directories.

**Comment by ttyva** [comment]
+1
options.sh is trying to start a jail with an empty name.  Manually specifying the target arch doesn't help.  This breaks the use case where options are set without a specific jail (by ports tree+set for us).

**Comment by fernape** [comment]
I've been bitten by this today.
 
It is broken regardless of the jail name. Without the name of the jail, it shows that weird error. With the name, it reports the port does not exist (but it does, and poudriere can built it).

```
pkg info poudriere-devel
poudriere-devel-3.4.99.20231113
```
```
$ sudo poudriere options -p default databases/sql-workbench
Password:
[00:00:00] Working on options directory: /usr/local/etc/poudriere.d/default-options
[00:00:00] Using ports from: /usr/local/poudriere/ports/default
[00:00:00] Appending to make.conf: /usr/local/etc/poudriere.d/make.conf
[00:00:00] Error: _jget: 3 arguments expected: var_return ptname property
```
```
$ sudo poudriere options -p default -j 13_2amd64 databases/sql-workbench
[00:00:00] Working on options directory: /usr/local/etc/poudriere.d/13_2amd64-default-options
[00:00:00] Using ports from: /usr/local/poudriere/ports/default
[00:00:00] Appending to make.conf: /usr/local/etc/poudriere.d/make.conf
[00:00:00] Creating the reference jail... done
[00:00:01] Mounting system devices for 13_2amd64-default
[00:00:01] Mounting ccache from: /var/cache/ccache
[00:00:01] Mounting ports from: /usr/local/poudriere/ports/default
[00:00:01] Mounting packages from: 
[00:00:01] Mounting distfiles from: /data/usr/ports/distfiles/
[00:00:01] Copying /var/db/ports from: /usr/local/etc/poudriere.d/13_2amd64-default-options
[00:00:01] Appending to make.conf: /usr/local/etc/poudriere.d/make.conf
/etc/resolv.conf -> /usr/local/poudriere/data/.m/13_2amd64-default/ref/etc/resolv.conf
[00:00:01] Starting jail 13_2amd64-default
[00:00:02] Ports supports: FLAVORS SELECTED_OPTIONS
[00:00:02] Error: No such port: databases/sql-workbench
```

**Comment by mfechner** [comment]
I have a similar problem, to "fix" it I downgraded from `poudriere-devel` to `poudriere`

**Comment by mvanbaak** [comment]
This was fixed in https://github.com/freebsd/poudriere/commit/8726f468060a8886cdfac6d20cae4c997f735406 (I updated the poudriere-devel port to this commit, and can confirm it is working correctly now)

**Comment by mattxtaz** [comment]
Doesn't seem to help me. This is actually the first thing I tried, as I saw this commit, but raised this issue after finding it didn't change anything:

> % fetch -o /usr/local/share/poudriere/options.sh https://raw.githubusercontent.com/freebsd/poudriere/8726f468060a8886cdfac6d20cae4c997f735406/src/share/poudriere/options.sh
> /usr/local/share/poudriere/options.sh                 7434  B 7831 kBps    00s
> % poudriere options -n -f /usr/local/etc/poudriere.d/optlist
> [00:00:00] Working on options directory: /usr/local/etc/poudriere.d/options
> [00:00:00] Using ports from: /usr/ports
> [00:00:00] Appending to make.conf: /usr/local/etc/poudriere.d/make.conf
> [00:00:00] Error: _jget: 3 arguments expected: var_return ptname property
>

**Comment by mattxtaz** [comment]
Correct. If I add -j 140amd64 then it works. So it's fixed one use case, but it should still work with an empty jail name as it did before the recent update.

**Comment by madpilot78** [comment]
Looks like there are two issues here. @mattxtaz  is using `poudriere options` without specifying a jail. This is something I did not foresee when I submitted my patch. Maybe it did work in the "old" versions, but the whole options.sh file was broken at some point and was failing to properly set options for flavored ports. My patch tried to address that. I have no idea how to fix the original bug and not make `poudriere options` require specifying a jail to start.

The other problem reported by @fernape  is a bug I introduced with my patch. Poudriere is truing to use a non existent ports tree, and is fixed by 8726f468060a8886cdfac6d20cae4c997f735406, as @mvanbaak observed.

@bapt The poudriere-devel port in the ports tree should be updated to a commit that includes 8726f468060a8886cdfac6d20cae4c997f735406 to avoid hitting that bug again via the official tree port.

I'm not sure what is your opinion about running `poudriere options` without specifying a jail. With my patch included this is simply not possible, and my opinion is that we should make specifying the jail mandatory. I don't know how to fix the `options.sh` logic without starting a jail to extract the required information from the ports tree.

**Comment by fernape** [comment]
> Looks like there are two issues here. @mattxtaz is using `poudriere options` without specifying a jail. This is something I did not foresee when I submitted my patch. Maybe it did work in the "old" versions, but the whole options.sh file was broken at some point and was failing to properly set options for flavored ports. My patch tried to address that. I have no idea how to fix the original bug and not make `poudriere options` require specifying a jail to start.
> 
> The other problem reported by @fernape is a bug I introduced with my patch. Poudriere is truing to use a non existent ports tree, and is fixed by [8726f46](https://github.com/freebsd/poudriere/commit/8726f468060a8886cdfac6d20cae4c997f735406), as @mvanbaak observed.
> 
> @bapt The poudriere-devel port in the ports tree should be updated to a commit that includes [8726f46](https://github.com/freebsd/poudriere/commit/8726f468060a8886cdfac6d20cae4c997f735406) to avoid hitting that bug again via the official tree port.
> 
> I'm not sure what is your opinion about running `poudriere options` without specifying a jail. With my patch included this is simply not possible, and my opinion is that we should make specifying the jail mandatory. I don't 

I would not oppose strongly to this, but if we want to test the same option combination in say 12.4amd64, 12.4i386, 13.2amd64 and 14.0amd64, that means 5 `puoudriere options` :-)

> know how to fix the `options.sh` logic without starting a jail to extract the required information from the ports tree.

**Comment by madpilot78** [comment]
> > I'm not sure what is your opinion about running `poudriere options` without specifying a jail. With my patch included this is simply not possible, and my opinion is that we should make specifying the jail mandatory. I don't
> 
> I would not oppose strongly to this, but if we want to test the same option combination in say 12.4amd64, 12.4i386, 13.2amd64 and 14.0amd64, that means 5 `puoudriere options` :-)
> 

Not necessarily. You can just symlink all option directories to the same one and just run options once, which is fundamentally the same you are already doing. You'd just have to choose with which jail to run the option.

On the other hand having poudriere pick a random jail, is prone to foot shooting.

Older method was to use the host OS to configure options, but that stopped working at some point, most probably with the addition of SUBPACKAGES support. It is also not completely correct IMHO, because a port could offer different options depending on the OS version, and using the base system is not correct in that case.

> > know how to fix the `options.sh` logic without starting a jail to extract the required information from the ports tree.

**Comment by mattxtaz** [comment]
If you read the man page for poudriere-options it doesn't include -j in the synopsis at the top, and in the description of -j it specifically states "if given" meaning it is optional. That's why I never used this option in the past. I only ever have one jail, so I assume it just automatically uses the only jail that exists.

I have no problem with always using the -j option to specify the jail, but if this is mandatory then the man page should be updated and also it should show the usage text rather than an error about a function only having 2 arguments rather than 3.

**Comment by madpilot78** [comment]
Bapt ha produced a much better fix to all these issues here: https://github.com/freebsd/poudriere/issues/1083#issuecomment-1848725388

**Comment by madpilot78** [comment]
This should have been fixed by 9143ab494773f7a1ba69efc31177d4bbfe51bf7f and imported in the ports tree, even in the stable port.

Can this issue be closed?

**Comment by mattxtaz** [comment]
Yes. I now get a warning that tells me to specify -j. So I consider that closed. Thanks!
